docs/madr/decisions/049-meshexternalservice-api.md

```diff
@@ -0,0 +1,694 @@
+# MeshExternalService API
+
+* Status: accepted
+
+Technical Story: https://github.com/kumahq/kuma/issues/6330
+
+## Context and Problem Statement
+
+The current implementation of `ExternalService` has significant limitations and warrants a complete redesign.
+
+* Lack of support for * subdomains
+* Lack of support for thick clients (think Kafka or Cassandra where discovery is done as part of the client).
+* Need for pluggability (for example lambda support or equivalent)
+* Working well with gateways (what if your ExternalService relies on SNI/host header to route through a gateway)
+* Applying policies with and without egress
+
+## Considered Options
+
+* Creating MeshExternalService resource and MeshPassthrough policy
+* Creating MeshExternalService
+* Creating MeshExternalService + HostnameGenerator
+* Creating MeshExternalService and using HostnameGenerator for InternalVIP
+
+## Decision Outcome
+
+Chosen option: "Creating MeshExternalService resource and MeshPassthrough policy".
+
+### Positive Consequences
+
+* Clearer structure.
+* Explicit showing what is passthough and what is external service
+* Enhanced functionalities.
+* Better policy matching capabilities.
+
+### Negative Consequences
+
+* 2 new resources
+* Temporarily increased complexity of a product until the migration is done.
+
+## Pros and Cons of the Options
+
+### Creating MeshExternalService
+
+We're introducing a new object resource `MeshExternalService`. This object will be a namespace resource and will only be creatable within the `kuma-system` namespace. It can be created within a specific zone, but in this case, it will only be accessible within the zone where it was created.
+
+```yaml
+kind: MeshExternalService
+metadata:
+  name: example
+  namespace: kuma-system
+  labels:
+    kuma.io/mesh: default
+    kuma.io/zone: east-1
+spec:
+  type: Managed # Managed
+  managed:
+    match:
+      type: KumaHostname # Kuma will generate a domain
```

Comment:
```suggestion
      type: KumaHostname # Kuma will generate a domain
```

Are there other types? Should we just ignore this type ftm?

Reply:
If we keep it should we name it just `HostnameGenerator` after all it's the kind used later.

Reply:
I think I'd keep the type and rename it to indicate the link between this and HostnameGenerator, WDYT?

Reply:
Ok